src/test/java/uk/nhs/hee/tis/trainee/sync/service/PersonCachingTest.java

```diff
@@ -0,0 +1,206 @@
+    Field field = ReflectionUtils.findField(TraineeDetailsMapperImpl.class, "traineeDetailsUtil");
+    assert field != null;
+    field.setAccessible(true);
+    ReflectionUtils.setField(field, mapper, new TraineeDetailsUtil());
+
+    restTemplate = mock(RestTemplate.class);
+    service = new TcsSyncService(restTemplate, mapper, personService);
+
+    record = new Person();
+    record.setTisId(ID);
+    record.setOperation(Operation.DELETE);
+    record.setTable(Person.ENTITY_NAME);
+
+    personCache = cacheManager.getCache(Person.ENTITY_NAME);
+  }
+
+  @Test
+  void shouldDeletePersonByIdFromRepository() {
+    personService.deleteById(ID);
+    verify(mockPersonRepository).deleteById(ID);
+    verifyNoMoreInteractions(mockPersonRepository);
+  }
+
+  @Test
+  void shouldUseCacheForSecondInvocationOfRecord() {
+
+    when(mockPersonRepository.findById(record.getTisId()))
+        .thenReturn(Optional.of(record), Optional.of(new Person()));
+    assertThat(personCache.get(record.getTisId())).isNull();
+
+    Optional<Person> actual1 = personService.findById(record.getTisId());
+    assertThat(personCache.get(record.getTisId())).isNotNull();
+    Optional<Person> actual2 = personService.findById(record.getTisId());
+
+    verify(mockPersonRepository).findById(record.getTisId());
+    assertThat(actual1).isPresent().get().isEqualTo(record)
+        .isEqualTo(actual2.orElseThrow());
+  }
+
+  @Test
+  void shouldEvictFromCacheAndMissCacheOnNextInvocation() {
+
+    Person otherPerson = new Person();
+    final String otherId = "30";
+    otherPerson.setTisId(otherId);
+    otherPerson.setOperation(Operation.LOAD);
+
+    when(mockPersonRepository.findById(otherId)).thenReturn(Optional.of(otherPerson));
+    service.findById(otherId);
+
+    assertThat(personCache.get(otherId)).isNotNull();
+    when(mockPersonRepository.findById(ID)).thenReturn(Optional.of(record));
+
+    service.findById(ID);
+
+    assertThat(personCache.get(ID)).isNotNull();
+
+    service.syncRecord(record);
+
+    assertThat(personCache.get(ID)).isNull();
+    assertThat(personCache.get(otherId)).isNotNull();
+
+    verify(mockPersonRepository).deleteById(ID);
+
+    service.findById(ID);
+    assertThat(personCache.get(ID)).isNotNull();
+
+    verify(mockPersonRepository, times(2)).findById(record.getTisId());
+  }
+
+  @Test
+  void shouldReplaceCacheWhenSaved() {
+
+    data = new HashMap<>();
+    data.put("surname", "oldSurname");
+    data.put("role", "DR in Training");
+
+    Person stalePerson = new Person();
+
+    stalePerson.setTisId(ID);
+    stalePerson.setTable(Person.ENTITY_NAME);
+    stalePerson.setOperation(Operation.UPDATE);
+    stalePerson.setData(data);
+
+    when(mockPersonRepository.save(stalePerson)).thenReturn(stalePerson);
+    service.syncRecord(stalePerson);
+    assertThat(personCache.get(ID).get()).isEqualTo(stalePerson);
+
+    Person updatePerson = new Person();
+    data.put("surname", "oldSurname");
```

Comment:
Pepe picked this up, I had missed it :)
You need to set this to something different to the original data.surname (i.e. != "oldSurname") to actually test that its been replaced correctly in the cache. 
If you want to explicitly check that the new surname has been updated in the cache you can also add an assert like

assertThat(((Person) personCache.get(ID).get()).getData().get("surname")).isEqualTo(data.get("surname"));